packages/cli/generators/relation/templates/controller-relation-template-has-many.ts.ejs

```diff
@@ -58,7 +58,7 @@ export class <%= controllerClassName %> {
     @requestBody({
       content: {
         'application/json': {
-          schema: getModelSchemaRef(<%= targetModelClassName %>, {exclude: ['<%= targetModelPrimaryKey %>']}),
+          schema: getModelSchemaRef(<%= targetModelClassName %>, {exclude: ['<%= targetModelPrimaryKey %>'], optional: ['<%= foreignKeyName %>']}),
```

Comment:
even putting the **foreignKeyName** in exclude list does the trick. Not sure how this new optional key helps?

Reply:
We can debate if `foreignKeyName` should be marked as `exclude` or `optional`. For example, to create an order for a customer, the FK of order is `customerId`. It is automatically populated/enforced.

I think mark it `optional` and make sure the value does not violate the constrain is reasonable.

Reply:
I think it makes sense to keep it as `optional` and allow the developer to choose whether or not to include it in the request body. 

Currently, if it is included and is different than the one passed in the endpoint, the user gets an error saying they're not allowed to change the foreign key value.

Reply:
I agree that `optional` makes more sense.
But I don't know if `exclude` should work as `optional`.

Reply:
@agnes512 @raymondfeng When we work through related models general behaviour as per REST conventions also the child is being created for the parent passed in request URL. For e.g.  if url is **customer/<customerId>/order** this means one is trying to create an order with the given customer id in URL. Here the context which we are talking is of the controller created by LB4 relation which is for relational model i.e. **CustomerOrderController** not the **OrderController**. So in this case customer_id which is the foreign key in order must have the same value as of URL.
Right now I have to manually go to the controller and write the code to fetch the ID from URL and attach it to request body which I believe should happen automatically.

Reply:
@bhupesh-sf I'm not sure if the LB runtime will have the knowledge to fetch the `customerId` from the url and attach it to the requestBody. What do you in mind to achieve that? BTW, the url can even skip `:customerId` as it can be inferred from the logged user.

The only (theoretical) possibility is to allow request body to have `variables` that can resolved from the context, such as:
```
{
   "customerId": ${user.id},
}
```